sherpa/data.py

```diff
@@ -1144,18 +1203,36 @@ def set_dep(self, val):
             If a number then it is used for each element.
 
         """
-        if numpy.iterable(val):
-            dep = numpy.asarray(val, SherpaFloat)
+        if np.iterable(val):
+            dep = np.asarray(val, SherpaFloat)
         else:
             nelem = self.size
             if nelem is None:
                 raise DataErr("sizenotset", self.name)
 
-            val = SherpaFloat(val)
-            dep = val * numpy.ones(nelem, dtype=SherpaFloat)
+            dep = SherpaFloat(val) * np.ones(nelem, dtype=SherpaFloat)
 
         self.y = dep
 
+    # It is not clear when default values need to be given in overload
+    # statements. The current choice is based on mypy 1.10.0.
+    #
+    @overload
+    def get_y(self,
+              filter: bool,
+              yfunc: None,
+              use_evaluation_space: bool = False
+              ) -> np.ndarray:
+        ...
+
+    @overload
+    def get_y(self,
```

Comment:
Again, beyond the scope of this PR, but if we designed this form scratch, we might make it two different functions instead of one functions with different signatures...